fota/fota.h

```diff
@@ -0,0 +1,62 @@
+// ----------------------------------------------------------------------------
+// Copyright 2018-2019 ARM Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------
+
+#ifndef __FOTA_H_
+#define __FOTA_H_
+
+#include "fota/fota_config.h"
+
+#if MBED_CLOUD_CLIENT_FOTA_ENABLE
+
+// TODO: move to delta -  when integrated
+#if !defined(MBED_CLOUD_CLIENT_FOTA_DELTA_BLOCK_SIZE)
+#define MBED_CLOUD_CLIENT_FOTA_DELTA_BLOCK_SIZE 1024
+#endif
+
+
+#include "fota/fota_status.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct endpoint_s endpoint_t;
```

Comment:
Why do you typedef something that is coming from the Client headers?
To be consistent I think we should be use the types as they are defined.

Reply:
this is forward declaration - the type is coming from client header files 
@michael - do we even need to import this file?

Reply:
Forward declaration makes sense for opaque types.. but I don't see why you changed the name as well.. however, this is minor detail.

Reply:
it was imported, but no more from latest commit